tests/contrib/grpc/test_grpc_utils.py

```diff
@@ -23,19 +22,72 @@ def test_parse_method_path_without_package():
     [
         (("localhost:1234",), dict(), ("localhost", 1234), None),
         (("localhost",), dict(), ("localhost", None), None),
+        ((":1234",), dict(), (None, 1234), None),
         (("[::]:1234",), dict(), ("::", 1234), None),
         (("[::]",), dict(), ("::", None), None),
         (None, dict(target="localhost:1234"), ("localhost", 1234), None),
         (None, dict(target="localhost"), ("localhost", None), None),
+        (None, dict(target=":1234"), (None, 1234), None),
         (None, dict(target="[::]:1234"), ("::", 1234), None),
         (None, dict(target="[::]"), ("::", None), None),
         (("localhost:foo",), dict(), ("localhost", None), ("Non-integer port in target '%s'", "localhost:foo")),
         (("",), dict(), (None, None), None),
     ],
 )
 def test_parse_target_from_args(mock_log, args, kwargs, result, log_warning_call):
-    assert _parse_target_from_args(args, kwargs) == result
+    assert utils._parse_target_from_args(args, kwargs) == result
     if log_warning_call:
         mock_log.warning.assert_called_once_with(*log_warning_call)
     else:
         mock_log.warning.assert_not_called()
+
+
+@pytest.mark.parametrize(
+    "host, port, calls",
+    [
+        (
+            "localhost",
+            1234,
+            [mock.call("grpc.host", "localhost"), mock.call("grpc.port", "1234"), mock.call("span.kind", "client")],
+        ),
+        ("localhost", None, [mock.call("grpc.host", "localhost"), mock.call("span.kind", "client")]),
+        (None, 1234, [mock.call("grpc.port", "1234"), mock.call("span.kind", "client")]),
+        (None, None, [mock.call("span.kind", "client")]),
+    ],
+)
+def test_set_grpc_client_meta(host, port, calls):
```

Comment:
I'm not a big fan of this pattern because it's a bit removed from what could happen realistically in the integration - but I also understand that it's hard to replicate these scenarios

Reply:
Right, I think in this case I opted to go the unit testing route rather than an integration test because it's unclear how we could reproduce such a scenario. We could spend some time moving forward with rethinking these tests though.